primitives/primproc/bppsendthread.cpp

```diff
@@ -254,19 +265,26 @@ void BPPSendThread::mainLoop()
                 (void)atomicops::atomicSub(&currentByteSize, bsSize);
                 msg[msgsSent].msg.reset();
             }
+
+            if (respondWait && currentByteSize < maxByteSize)
+            {
+                okToRespond.notify_one();
+            }
         }
     }
 }
 
 void BPPSendThread::abort()
 {
-    boost::mutex::scoped_lock sl(msgQueueLock);
-    boost::mutex::scoped_lock sl2(ackLock);
-    die = true;
-    queueNotEmpty.notify_one();
-    okToSend.notify_one();
-    sl.unlock();
-    sl2.unlock();
+    {
+        std::lock_guard<std::mutex> sl(msgQueueLock);
```

Comment:
I was curios about the lifetime of these lock_guards and it seems their dtors are called exiting the scope at line 284 not in the end of the function. [Here](http://cpp.sh/2xqv3) is a fiddler to see and run the code. Is this an expected behavior?

Reply:
This is the expected behavior. The lock is only taken if SizeTooBig -- a rare case. It's there to protect the condition variable. Once we're done with the condition, then we don't need or want the lock.